### DISCUSSION TITLE
get the editor vim mode through API

### DISCUSSION THREAD

**Discussion by Zane-XY** [JetBrains/ideavim#695] [Q&A] [answered]
I'm working on a plugin, which needs to know if the current editor is in Vim's insert mode. For example, my plugin provides a Tab completion feature, which should only be triggered if the user is in Vim's insert mode. I need to know if the user is currently using Vim, and what is the corresponding mode for the current editor if Vim is enabled.

**Comment by AlexPl292** [comment, marked as answer]
Hi, please check this answer: https://youtrack.jetbrains.com/issue/VIM-3057/Get-the-current-VimEditor-mode-through-Intellij-platform-API#focus=Comments-27-8046185.0-0
